src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -65,6 +72,47 @@ private void AddExceptionDiagnostic(Diagnostic diagnostic)
         }
 
         /// <summary>
+        /// Returns diagnostics produced by diagnostic analyzers from analyzing a single document.
```

Comment:
The comment has a sort of low-quality correctness, but I admit and agree that it is significantly misleading and needs to change. I'm too tired to come up with a decent wording tonight.